safe-global/safe-wallet-web
# [Counterfactual] Add pay now pay later option to safe creation

## What it solves

Resolves #3210 

## How this PR fixes it

- Adds a Pay now / Pay later option to the safe creation flow ONLY for counterfactual safes

## How to test it

1. Create a new 1/1 safe
2. Observe a Pay now / Pay later option on the review screen
3. Choose Pay later and create
4. You should immediately see the dashboard. Remember the safe address
5. Go through the safe creation flow again
6. Choose Pay later and either choose relayer or pay via wallet
7. Observe the transaction is being executed and the same safe address is deployed
8. Open the safe creation flow again and choose a different setup e.g. 1/2
9. Observe no Pay now / Pay later option
10. The safe should be created as normal

## Screenshots

Counterfactual without relaying:
<img width="807" alt="Screenshot 2024-02-10 at 11 54 48" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/ca597312-1ad1-4565-9e48-d39b3a470013">

Counterfactual with relaying:
<img width="796" alt="Screenshot 2024-02-10 at 11 55 12" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/f1cb5290-69d4-4177-bb71-3876ef24e17e">

No counterfactual with relaying:
<img width="790" alt="Screenshot 2024-02-10 at 11 55 31" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/fe308179-7511-4169-adc4-ea0a88122278">

No counterfactual without relaying:
<img width="799" alt="Screenshot 2024-02-10 at 11 55 50" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/c8b86bed-a0b8-40f2-946c-0bb05f9ee267">

## Checklist
